### PR TITLE
Add support for Heroku

### PR DESCRIPTION
**Pull request by codeinteger6**
Made the bot compatible with Heroku, and added a button on the readme for easy deployment.

### CODE REVIEW

**Comment by pmariglia** [comment]
Hey, I appreciate the effort but I'm a bit confused about the desire to have this deployed on Heroku

As this is a command-line application, what is the benefit from deploying it? The Heroku application page wouldn't show anything when connecting and the bot would just run in the background.

File: app.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "Showdown bot",
+  "description": "A Pokemon Showdown Battle Bot written in Python",
+  "repository": "https://github.com/pmariglia/showdown",
+  "env": {
+    "WEBSOCKET_URI": {
+      "description": "The address to use to connect to the Pokemon Showdown websocket",
+      "value": "sim.smogon.com:8080",
```
Comment (pmariglia) [inline app.json]: looks like a typo - the PokemonShowdown websocket server runs on port 8000
```suggestion
      "value": "sim.smogon.com:8000",
```

**Comment by codeinteger6** [comment]
> Hey, I appreciate the effort but I'm a bit confused about the desire to have this deployed on Heroku
> 
> As this is a command-line application, what is the benefit from deploying it? The Heroku application page wouldn't show anything when connecting and the bot would just run in the background.

It would run as a worker not a web application, it's just another good site to be able to host the bot and making the bot ready for deployment would make things easier for people who would want to run it on Heroku. I'm pretty sure there may be some people out there who would want to use Heroku. And the bot can be switched off when it's not needed.

**Comment by pmariglia** [comment]
> It would run as a worker not a web application, it's just a good site to be able to host the bot for free and making the bot ready for deployment would make things easier for people who would want to run it on Heroku. I'm pretty sure there may be some people out there who would want to use Heroku. And the bot can be switched off when it's not needed.

I see. I suppose this could be useful for people wanting the bot to run quickly.

I've thought about having a version of the bot up 24/7 accepting challenges from people on the official PokemonShowdown server. Heroku would allow me to do that.

**Pull request merged**